update the plugin?

im using 1.21.11 (i already tested in 1.21.8 and 1.21.4) with towny 0.102.0.3 and towny combat gives this error in console

[21:56:21 INFO]: [TownyCombat] Towny version 0.102.0.3 found.
[21:56:21 ERROR]: Error occurred while enabling TownyCombat v4.2.1 (Is it up to date?)
java.lang.NoClassDefFoundError: com/palmergames/adventure/text/event/HoverEventSource
        at TownyCombat-4.2.1(1).jar//io.github.townyadvanced.townycombat.TownyCombat.registerListeners(TownyCombat.java:106) ~[?:?]
        at TownyCombat-4.2.1(1).jar//io.github.townyadvanced.townycombat.TownyCombat.onEnable(TownyCombat.java:45) ~[?:?]
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:279) ~[leaf-api-1.21.11-R0.1-SNAPSHOT.jar:?]
        at io.papermc.paper.plugin.manager.PaperPluginInstanceManager.enablePlugin(PaperPluginInstanceManager.java:202) ~[leaf-1.21.11.jar:1.21.11-20-9e82aaa]
        at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.enablePlugin(PaperPluginManagerImpl.java:109) ~[leaf-1.21.11.jar:1.21.11-20-9e82aaa]
        at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:519) ~[leaf-api-1.21.11-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.craftbukkit.CraftServer.enablePlugin(CraftServer.java:656) ~[leaf-1.21.11.jar:1.21.11-20-9e82aaa]
        at org.bukkit.craftbukkit.CraftServer.enablePlugins(CraftServer.java:613) ~[leaf-1.21.11.jar:1.21.11-20-9e82aaa]
        at net.minecraft.server.MinecraftServer.initPostWorld(MinecraftServer.java:689) ~[leaf-1.21.11.jar:1.21.11-20-9e82aaa]
        at net.minecraft.server.dedicated.DedicatedServer.initServer(DedicatedServer.java:426) ~[leaf-1.21.11.jar:1.21.11-20-9e82aaa]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1334) ~[leaf-1.21.11.jar:1.21.11-20-9e82aaa]
        at net.minecraft.server.MinecraftServer.lambda$spin$2(MinecraftServer.java:429) ~[leaf-1.21.11.jar:1.21.11-20-9e82aaa]
        at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]
Caused by: java.lang.ClassNotFoundException: com.palmergames.adventure.text.event.HoverEventSource
        at org.bukkit.plugin.java.PluginClassLoader.loadClass0(PluginClassLoader.java:208) ~[leaf-api-1.21.11-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.java.PluginClassLoader.loadClass(PluginClassLoader.java:175) ~[leaf-api-1.21.11-R0.1-SNAPSHOT.jar:?]
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526) ~[?:?]
        ... 13 more
[21:56:21 INFO]: [TownyCombat] Disabling TownyCombat v4.2.1

--- a/pom.xml
+++ b/pom.xml
@@ -9,7 +9,7 @@
     <java.version>17</java.version>
     <project.bukkitAPIVersion>1.15</project.bukkitAPIVersion>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <towny.version>0.101.0.2</towny.version>
+    <towny.version>0.102.0.0</towny.version>
   </properties>
 
   <repositories>
@@ -18,8 +18,8 @@
       <url>https://repo.glaremasters.me/repository/towny/</url>
       </repository>    
     <repository>
-      <id>spigot-repo</id>
-      <url>https://hub.spigotmc.org/nexus/content/repositories/snapshots/</url>
+      <id>papermc</id>
+      <url>https://repo.papermc.io/repository/maven-public/</url>
     </repository>
 	<repository>
 	  <id>jitpack.io</id>
@@ -33,9 +33,9 @@
 
   <dependencies>
     <dependency>
-      <groupId>org.spigotmc</groupId>
-      <artifactId>spigot-api</artifactId>
-      <version>1.16.5-R0.1-SNAPSHOT</version>
+      <groupId>io.papermc.paper</groupId>
+      <artifactId>paper-api</artifactId>
+      <version>1.21.10-R0.1-SNAPSHOT</version>
       <scope>provided</scope>
     </dependency>
 	<dependency>

--- a/src/main/java/io/github/townyadvanced/townycombat/TownyCombat.java
+++ b/src/main/java/io/github/townyadvanced/townycombat/TownyCombat.java
@@ -24,7 +24,7 @@ public class TownyCombat extends JavaPlugin {
 	
 	private static TownyCombat plugin = null;
 	private static DynmapIntegration dynmapIntegration = null;
-	private static final Version requiredTownyVersion = Version.fromString("0.101.0.0");
+	private static final Version requiredTownyVersion = Version.fromString("0.102.0.0");
 	private static BukkitTask townyCombatTask = null;
 
     @Override

--- a/src/main/java/io/github/townyadvanced/townycombat/listeners/TownyCombatBukkitEventListener.java
+++ b/src/main/java/io/github/townyadvanced/townycombat/listeners/TownyCombatBukkitEventListener.java
@@ -24,6 +24,8 @@ import org.bukkit.event.Listener;
 import org.bukkit.event.entity.EntityDamageByEntityEvent;
 import org.bukkit.event.entity.EntityDamageEvent;
 import org.bukkit.event.entity.EntityDamageEvent.DamageCause;
+import org.bukkit.event.entity.EntityDismountEvent;
+import org.bukkit.event.entity.EntityMountEvent;
 import org.bukkit.event.entity.EntityShootBowEvent;
 import org.bukkit.event.entity.PotionSplashEvent;
 import org.bukkit.event.inventory.BrewEvent;
@@ -38,8 +40,8 @@ import org.bukkit.inventory.BrewerInventory;
 import org.bukkit.inventory.ItemStack;
 import org.bukkit.inventory.meta.PotionMeta;
 import org.bukkit.potion.PotionEffectType;
-import org.spigotmc.event.entity.EntityDismountEvent;
-import org.spigotmc.event.entity.EntityMountEvent;
+//import org.spigotmc.event.entity.EntityDismountEvent;
+//import org.spigotmc.event.entity.EntityMountEvent;
 
 /**
  * 
@@ -160,7 +162,7 @@ public class TownyCombatBukkitEventListener implements Listener {
 				event.setCancelled(true);
 				AbstractHorse horse = (AbstractHorse)event.getEntity();
 				double newHealth = horse.getHealth() - event.getDamage();
-				horse.setHealth(Math.max(0, Math.min(horse.getAttribute(Attribute.GENERIC_MAX_HEALTH).getValue(), newHealth)));
+				horse.setHealth(Math.max(0, Math.min(horse.getAttribute(Attribute.MAX_HEALTH).getValue(), newHealth)));
 			}
 		}
 	}
@@ -309,7 +311,7 @@ public class TownyCombatBukkitEventListener implements Listener {
 			(attackingPlayer != null && attackingPlayer.isInsideVehicle() && attackingPlayer.getVehicle() instanceof AbstractHorse) 				
 			&&
 			//Bonus is charged-up
-			(attackingPlayer.hasPotionEffect(PotionEffectType.INCREASE_DAMAGE) && attackingPlayer.getPotionEffect(PotionEffectType.INCREASE_DAMAGE).getAmplifier() == -1))
+			(attackingPlayer.hasPotionEffect(PotionEffectType.STRENGTH) && attackingPlayer.getPotionEffect(PotionEffectType.STRENGTH).getAmplifier() == -1))
 		{
 			TownyCombatSpecialCavalryHitEvent specialCavalryHit = new TownyCombatSpecialCavalryHitEvent(attackingPlayer, event.getEntity(), true);
 			Bukkit.getPluginManager().callEvent(specialCavalryHit);

--- a/src/main/java/io/github/townyadvanced/townycombat/listeners/TownyCombatStatusScreenListener.java
+++ b/src/main/java/io/github/townyadvanced/townycombat/listeners/TownyCombatStatusScreenListener.java
@@ -1,8 +1,5 @@
 package io.github.townyadvanced.townycombat.listeners;
 
-
-import com.palmergames.adventure.text.Component;
-import com.palmergames.adventure.text.event.HoverEvent;
 import com.palmergames.bukkit.towny.event.statusscreen.ResidentStatusScreenEvent;
 import com.palmergames.bukkit.towny.object.Resident;
 import com.palmergames.bukkit.towny.object.Translation;
@@ -12,6 +9,9 @@ import io.github.townyadvanced.townycombat.TownyCombat;
 import io.github.townyadvanced.townycombat.events.BattlefieldRole;
 import io.github.townyadvanced.townycombat.settings.TownyCombatSettings;
 import io.github.townyadvanced.townycombat.utils.TownyCombatBattlefieldRoleUtil;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.event.HoverEvent;
+
 import org.bukkit.event.EventHandler;
 import org.bukkit.event.Listener;
 

--- a/src/main/java/io/github/townyadvanced/townycombat/utils/TownyCombatBattlefieldRoleUtil.java
+++ b/src/main/java/io/github/townyadvanced/townycombat/utils/TownyCombatBattlefieldRoleUtil.java
@@ -328,7 +328,7 @@ public class TownyCombatBattlefieldRoleUtil {
             if (grantAdjustedPotionSpeedEffectToPlayer(originalPotionEffect, event.getPlayer())) {
                 event.setItem(new ItemStack(Material.GLASS_BOTTLE));
             }
-        } else if (potionEffectType.equals(PotionEffectType.INCREASE_DAMAGE)) {
+        } else if (potionEffectType.equals(PotionEffectType.STRENGTH)) {
             int potionAmplifier = potionData.isUpgraded() ? 1 : 0;
             int potionDurationTicks = (potionData.isExtended() ? 480 : potionData.isUpgraded() ? 90 : 180) * 20;
             PotionEffect originalPotionEffect = new PotionEffect(potionEffectType, potionDurationTicks, potionAmplifier, true, true, true);
@@ -377,7 +377,7 @@ public class TownyCombatBattlefieldRoleUtil {
                         }
                     }
                 }
-            } else if (potionEffect.getType().equals(PotionEffectType.INCREASE_DAMAGE)) {
+            } else if (potionEffect.getType().equals(PotionEffectType.STRENGTH)) {
                 for (LivingEntity entity : event.getAffectedEntities()) {
                     if (entity instanceof Player) {
                         if (event.getIntensity(entity) != 0 && grantAdjustedPotionStrengthEffectToPlayer(potionEffect, (Player) entity)) {
@@ -465,7 +465,7 @@ public class TownyCombatBattlefieldRoleUtil {
     
     private static void giveRoleBasedDamageResistance(LivingEntity livingEntity) {
         final int effectDurationTicks = (int) (TimeTools.convertToTicks(TownySettings.getShortInterval() + 10));
-        givePotionEffectUnlessAmplifierIsNegative(PotionEffectType.DAMAGE_RESISTANCE, effectDurationTicks,0, false, false, true, livingEntity);
+        givePotionEffectUnlessAmplifierIsNegative(PotionEffectType.RESISTANCE, effectDurationTicks,0, false, false, true, livingEntity);
     }
 
     private static void giveEffectUnlessAmplifierIsNegative(PotionEffectType potionEffectType, int durationTicks, int amplifier, LivingEntity livingEntity) {

--- a/src/main/java/io/github/townyadvanced/townycombat/utils/TownyCombatHorseUtil.java
+++ b/src/main/java/io/github/townyadvanced/townycombat/utils/TownyCombatHorseUtil.java
@@ -69,12 +69,12 @@ public class TownyCombatHorseUtil {
 
     public static void registerPlayerForCavalryStrengthBonus(Player player) {
         cavalryStrengthBonusRefreshTimes.put(player, System.currentTimeMillis() + TownyCombatSettings.getCavalryChargeCooldownMilliseconds());
-        player.removePotionEffect(PotionEffectType.INCREASE_DAMAGE); //Removed so as not to cause confusion, and/or in preparation for next hit if applicable
+        player.removePotionEffect(PotionEffectType.STRENGTH); //Removed so as not to cause confusion, and/or in preparation for next hit if applicable
     }
 
     public static void deregisterPlayerForCavalryStrengthBonus(Player player) {
         cavalryStrengthBonusRefreshTimes.remove(player);
-        player.removePotionEffect(PotionEffectType.INCREASE_DAMAGE); //Remove so as not to cause confusion
+        player.removePotionEffect(PotionEffectType.STRENGTH); //Remove so as not to cause confusion
     }
 
     public static boolean isHorseTeleportScheduled(AbstractHorse horse) {
@@ -106,7 +106,7 @@ public class TownyCombatHorseUtil {
      * This has no effect on its own, but is used to indicate that the strength bonus is ready for use.
      */
     private static void applyPlaceholderStrengthEffectToPlayer(Player player, int effectDurationTicks) {
-        player.addPotionEffect(new PotionEffect(PotionEffectType.INCREASE_DAMAGE, effectDurationTicks,-1));
+        player.addPotionEffect(new PotionEffect(PotionEffectType.STRENGTH, effectDurationTicks,-1));
     }
 
     public static @Nullable Player getPlayerRider(Entity entity) {
@@ -127,7 +127,7 @@ public class TownyCombatHorseUtil {
 
     public static void cancelStrengthEffectsOnPlayerRiders(PotionSplashEvent event) {
         for (PotionEffect potionEffect : event.getPotion().getEffects()) {
-            if (potionEffect.getType().equals(PotionEffectType.INCREASE_DAMAGE)) {
+            if (potionEffect.getType().equals(PotionEffectType.STRENGTH)) {
                 for (LivingEntity entity : event.getAffectedEntities()) {
                     if (entity instanceof Player && getMount((Player)entity) != null) {
                         event.setIntensity(entity, 0);
@@ -146,7 +146,7 @@ public class TownyCombatHorseUtil {
             PotionEffectType potionEffectType = potionMeta.getBasePotionData().getType().getEffectType();
             if(potionEffectType == null)
                 return;
-            if(potionEffectType.equals(PotionEffectType.INCREASE_DAMAGE)
+            if(potionEffectType.equals(PotionEffectType.STRENGTH)
                     && TownyCombatHorseUtil.getMount(event.getPlayer()) != null) {
                 event.setCancelled(true);
                 Messaging.sendMsg(event.getPlayer(), Translatable.of("msg_warning_potion_strength_effect_blocked_for_rider"));

--- a/src/main/java/io/github/townyadvanced/townycombat/utils/TownyCombatItemUtil.java
+++ b/src/main/java/io/github/townyadvanced/townycombat/utils/TownyCombatItemUtil.java
@@ -87,7 +87,7 @@ public class TownyCombatItemUtil {
         ItemMeta itemMeta = result.getItemMeta();
         itemMeta.setDisplayName(Translatable.of("spear_name").translate(Locale.ROOT));
         //Add enchants
-        itemMeta.addEnchant(Enchantment.DAMAGE_ALL, NATIVE_SPEAR_SHARPNESS_LEVEL, true);
+        itemMeta.addEnchant(Enchantment.SHARPNESS, NATIVE_SPEAR_SHARPNESS_LEVEL, true);
         //Add lore
         List<String> lore = new ArrayList<>();
         lore.add("");
@@ -235,7 +235,7 @@ public class TownyCombatItemUtil {
     }
 
     public static ItemStack createLingeringHarmPotion(Player owner, int durationSeconds, int amplifier) {
-        return createPotion(owner, Material.LINGERING_POTION, PotionEffectType.HARM, durationSeconds, amplifier, true, true, true, "super_potion_name_lingering_harm");
+        return createPotion(owner, Material.LINGERING_POTION, PotionEffectType.INSTANT_DAMAGE, durationSeconds, amplifier, true, true, true, "super_potion_name_lingering_harm");
     }
 
     public static ItemStack createAbsorbtionPotion(Player owner, int durationSeconds, int amplifier) {
@@ -294,7 +294,7 @@ public class TownyCombatItemUtil {
         for(ItemStack itemStack: player.getInventory().getContents()) {
             if(itemStack != null && (itemStack.getType() == Material.POTION || itemStack.getType() == Material.SPLASH_POTION)) {
                 oldPotionMeta = (PotionMeta) itemStack.getItemMeta();
-                if (oldPotionMeta != null && oldPotionMeta.getBasePotionData().getType() == PotionType.INSTANT_HEAL) {
+                if (oldPotionMeta != null && oldPotionMeta.getBasePotionData().getType() == PotionType.HEALING) {
                     //Create a new healing potion itemstack
                     ItemStack transmutedPotion = createTransmutedPotion(itemStack);
                     //Set the meta into the existing inventory item

--- a/src/main/java/io/github/townyadvanced/townycombat/utils/TownyCombatMovementUtil.java
+++ b/src/main/java/io/github/townyadvanced/townycombat/utils/TownyCombatMovementUtil.java
@@ -18,23 +18,23 @@ public class TownyCombatMovementUtil {
 
     public static void removeTownyCombatMovementModifiers(LivingEntity livingEntity) {
         //Remove the TCM modifiers
-        for (AttributeModifier attributeModifier : new ArrayList<>(livingEntity.getAttribute(Attribute.GENERIC_MOVEMENT_SPEED).getModifiers())) {
+        for (AttributeModifier attributeModifier : new ArrayList<>(livingEntity.getAttribute(Attribute.MOVEMENT_SPEED).getModifiers())) {
             if (attributeModifier.getName().equals(ATTRIBUTE_MODIFIER_NAME)) {
-                livingEntity.getAttribute(Attribute.GENERIC_MOVEMENT_SPEED).removeModifier(attributeModifier);
+                livingEntity.getAttribute(Attribute.MOVEMENT_SPEED).removeModifier(attributeModifier);
             }
         }
     }
 
     public static void removeTownyCombatKnockbackModifiers(LivingEntity livingEntity) {
-        for(AttributeModifier attributeModifier: new ArrayList<>(livingEntity.getAttribute(Attribute.GENERIC_KNOCKBACK_RESISTANCE).getModifiers())) {
+        for(AttributeModifier attributeModifier: new ArrayList<>(livingEntity.getAttribute(Attribute.KNOCKBACK_RESISTANCE).getModifiers())) {
             if(attributeModifier.getName().equals(ATTRIBUTE_MODIFIER_NAME)) {
-                livingEntity.getAttribute(Attribute.GENERIC_KNOCKBACK_RESISTANCE).removeModifier(attributeModifier);
+                livingEntity.getAttribute(Attribute.KNOCKBACK_RESISTANCE).removeModifier(attributeModifier);
             }
         }
     }
 
     public static void resetPlayerBaseSpeedToVanilla(Player player) {
-        player.getAttribute(Attribute.GENERIC_MOVEMENT_SPEED).setBaseValue(VANILLA_PLAYER_MOVEMENT_SPEED);        
+        player.getAttribute(Attribute.MOVEMENT_SPEED).setBaseValue(VANILLA_PLAYER_MOVEMENT_SPEED);        
     }
     
 }

